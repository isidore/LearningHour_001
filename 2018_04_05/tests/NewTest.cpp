#include "ApprovalTests.hpp"
#include "Catch.hpp"
#include <vector>


static string convert(int i) {


    if(i % 15 == 0)
        return "FizzBuzz";
    else if(i % 3 == 0)
        return "Fizz";
    else if (i % 5 == 0)
        return "Buzz";
    else
        return std::to_string(i);
}

TEST_CASE("FizzBuzzStep1")
{
    REQUIRE(convert(1) == "1");
}

TEST_CASE("FizzBuzzStep3")
{
    REQUIRE(convert(3) == "Fizz");
}

TEST_CASE("FizzBuzzStep5")
{
    REQUIRE(convert(5) == "Buzz");
}

TEST_CASE("FizzBuzzStep6")
{
    REQUIRE(convert(6) == "Fizz");
}

TEST_CASE("FizzBuzzStep9")
{
    REQUIRE(convert(9) == "Fizz");
}

TEST_CASE("FizzBuzzStep10")
{
    REQUIRE(convert(10) == "Buzz");
}

TEST_CASE("FizzBuzzStep15")
{
    REQUIRE(convert(15) == "FizzBuzz");
}
TEST_CASE("FizzBuzz")
{

    std::vector<int> v{1,3,5,6,9,10,15};
    Approvals::verifyAll("FIRST LETTER",v.begin(), v.end(), [](auto s, auto& os){os << s << " => " << convert(s);});
}
