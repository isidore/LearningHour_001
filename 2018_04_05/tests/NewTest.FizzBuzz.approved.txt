FIRST LETTER


1 => 1
3 => Fizz
5 => Buzz
6 => Fizz
9 => Fizz
10 => Buzz
15 => FizzBuzz

